### PR TITLE
Move and lint ECMA-335 Augments spec 

### PR DESCRIPTION
**Pull request by jnm2**
Followup from https://github.com/dotnet/csharplang/issues/2608#issuecomment-614768138:

> Yes, I think it would be fine to start a ECMA-335 Augments doc where we will collect additions to ECMA-335 we would like to do, but do not have a good way to do currently. We have number of those.
>
> Actually, we have a doc like already: https://github.com/dotnet/runtime/blob/master/src/libraries/System.Reflection.Metadata/specs/Ecma-335-Issues.md . We should build on top of it. Rename it ...Augments, and maybe move this whole directory to a more discoverable place, e.g. docs\design\specs.

Once this PR is merged, I will start a PR to add a [new section](https://github.com/dotnet/csharplang/issues/2608#issuecomment-614749417). The new section will need to be considered an official runtime specification and vetted by the runtime team. (See https://github.com/dotnet/roslyn/pull/43281#issuecomment-614791491.)

There are 11 remaining markdownlint warnings. Any or all of the linting commits can be dropped, just ask.

/cc @jkotas

### CODE REVIEW

File: docs/design/specs/Ecma-335-Augments.md [hunk 1]
```diff
@@ -1,4 +1,4 @@
-# Known Issues in ECMA-335 CLI Specification 
+# Additions to the ECMA-335 CLI Specification
 
 This is an informal list of notes on issues that have been encountered
 with the ECMA-335 CLI specification, primarily during development,
```
Comment (jnm2) [inline docs/design/specs/Ecma-335-Augments.md:4]: @jkotas Given that Aleksey is looking for a vetted runtime spec for module initializers, would you suggest any changes to this title and the introductory paragraphs yet?

Comment (jkotas) [reply docs/design/specs/Ecma-335-Augments.md:4]: What about:

This is a list of additions and edits to be made in ECMA-335 specifications. It includes both documentation of new runtime features and issues encountered during development.

**Comment by jkotas** [comment]
I would move all docs from https://github.com/dotnet/runtime/blob/master/src/libraries/System.Reflection.Metadata/specs/ to this directory

File: docs/design/specs/Ecma-335-Augments.md [hunk 2]
```diff
@@ -1,4 +1,4 @@
-# Known Issues in ECMA-335 CLI Specification 
+# Additions to the ECMA-335 CLI Specification
```
Comment (jkotas) [inline docs/design/specs/Ecma-335-Augments.md:1]: https://github.com/dotnet/runtime/blob/master/src/libraries/System.Reflection.Metadata/specs/PE-COFF.md title says "Addendum". Maybe we can do the same here to maintain same style.

**Comment by jkotas** [comment]
Add a link to this doc from https://github.com/dotnet/runtime/blob/master/docs/project/dotnet-standards.md#ecma-335---cli ?

**Comment by jnm2** [comment]
@jkotas Thanks for the help. All addressed!

**Review by jkotas — APPROVED** [review]

**Comment by jkotas** [comment]
Thank you!

**Pull request merged**